From af717260c43d798e12367bbcf31d2134b3f6c4e7 Mon Sep 17 00:00:00 2001
From: Simon Xue <xxm@rock-chips.com>
Date: Wed, 16 Jan 2019 14:19:03 +0800
Subject: [PATCH] PCI: rockchip: disable RIE in DMA Write Channel Control 1
 Register

After comparing the test results with RIE enabled and disabled,enable
RIE may cause PCIe dma interrupt work unnormal.

More works need to dig it out

Change-Id: Ia25022332de92cf21287df155cd514a160d140bc
Signed-off-by: Simon Xue <xxm@rock-chips.com>
---
 drivers/pci/host/rockchip-pcie-dma.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/pci/host/rockchip-pcie-dma.c b/drivers/pci/host/rockchip-pcie-dma.c
index 9f7fe96d2a01..ba8f86af2b15 100644
--- a/drivers/pci/host/rockchip-pcie-dma.c
+++ b/drivers/pci/host/rockchip-pcie-dma.c
@@ -221,7 +221,7 @@ static void rk_pcie_prepare_dma(struct dma_trx_obj *obj,
 	} else {
 		table->wr_enb.enb = 0x1;
 		table->ctx_reg.ctrllo.lie = 0x1;
-		table->ctx_reg.ctrllo.rie = 0x1;
+		table->ctx_reg.ctrllo.rie = 0x0;
 		table->ctx_reg.ctrllo.td = 0x1;
 		table->ctx_reg.ctrlhi.asdword = 0x0;
 		table->ctx_reg.xfersize = buf_size;
-- 
2.35.3

